native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java

```diff
@@ -154,73 +152,91 @@ private static Object convertElement(BXmlItem xmlItem, String attributePrefix,
             } else if (children == null) {
                 putAsBStrings(rootNode, keyValue, childrenData);
             } else if (children instanceof BString) {
-                putAsFieldTypes(childrenData, CONTENT, children.toString().trim(), type, uKeyValue);
+                putAsFieldTypes(childrenData, CONTENT, children.toString().trim(), getFieldType(keyValue, type));
                 putAsBStrings(rootNode, keyValue, childrenData);
                 return rootNode;
             }
         } else {
             if (children instanceof BMap) {
                 putAsBStrings(rootNode, keyValue, children);
             } else if (children == null) {
-                putAsFieldTypes(rootNode, keyValue, EMPTY_STRING, type, uKeyValue);
+                putAsFieldTypes(rootNode, keyValue, EMPTY_STRING, getFieldType(keyValue, type));
             } else if (children instanceof BString) {
-                putAsFieldTypes(rootNode, keyValue, children.toString().trim(), type, uKeyValue);
+                putAsFieldTypes(rootNode, keyValue, children.toString().trim(), getFieldType(keyValue, type));
             }
         }
         return rootNode;
     }
 
-    private static void processAttributes(BXmlItem xmlItem, String attributePrefix, BMap<BString, Object> mapData,
-                                          AttributeManager attributeManager, Type type, String uniqueKey,
+    private static Type getFieldType(String fieldName, Type type) {
+        if (type != null) {
+            if (type.toString().contains("record {| anydata...; |}")) {
```

Comment:
What is being checked here?. Is there any other way rather than having a string search (`"record {| anydata...; |}"`)

Reply:
If the expected record type is the default, we don't need to check the field of type. So, this is checking whether the expected type is default or not.

Reply:
Remove toString() [9df7eab](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/422/commits/9df7eab2448c1e0f5f33caa0f1e8783635d25d4e)

Reply:
Removed default type check [183c649](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/422/commits/183c649dd64d0ebbe5abf2c9c3b5874e88ea9a9a)